lib/src/migrators/module.dart

```diff
@@ -578,17 +678,23 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
     if (node.span.sourceUrl == _currentUrl) return null;
     if (!_namespaces.containsKey(node.span.sourceUrl)) {
       /// Add new use rule for indirect dependency
-      var relativePath = p.relative(node.span.sourceUrl.path,
-          from: p.dirname(_currentUrl.path));
-      var basename = p.basenameWithoutExtension(relativePath);
-      if (basename.startsWith('_')) basename = basename.substring(1);
-      var simplePath = p.relative(p.join(p.dirname(relativePath), basename));
+      var simplePath = _absoluteUriToDependency(node.span.sourceUrl);
       _additionalUseRules.add(simplePath);
       _namespaces[node.span.sourceUrl] = namespaceForPath(simplePath);
     }
     return _namespaces[node.span.sourceUrl];
   }
 
+  /// Converts an absolute URI for a stylesheet into the simplest string that
+  /// could be used to depend on that stylesheet from the current one in a use,
+  /// forward, or import rule.
+  String _absoluteUriToDependency(Uri uri) {
+    var relativePath = p.relative(uri.path, from: p.dirname(_currentUrl.path));
```

Comment:
```suggestion
    var relativePath = p.url.relative(uri.path, from: p.url.dirname(_currentUrl.path));
```

When you're working with URLs, you always want to use [`p.url`](https://pub.dev/documentation/path/latest/path/url.html). Otherwise you're interpreting URLs as native paths on whatever the current OS is, which will lead to all sorts of nasty edge-cases. Same throughout this function.